contracts/sg-ics721/src/testing/integration_tests.rs

```diff
@@ -91,68 +95,205 @@ fn no_init(
 ) {
 }
 
-#[derive(Debug)]
-struct Bech32AddressGenerator {
-    pub hrp: String,
+#[derive(Default)]
+pub struct MockAddressGenerator;
+
+impl AddressGenerator for MockAddressGenerator {
+    fn contract_address(
+        &self,
+        api: &dyn Api,
+        _storage: &mut dyn Storage,
+        code_id: u64,
+        instance_id: u64,
+    ) -> Result<Addr> {
+        let canonical_addr = Self::instantiate_address(code_id, instance_id);
+        Ok(Addr::unchecked(api.addr_humanize(&canonical_addr)?))
+    }
+
+    fn predictable_contract_address(
+        &self,
+        api: &dyn Api,
+        _storage: &mut dyn Storage,
+        _code_id: u64,
+        _instance_id: u64,
+        checksum: &[u8],
+        creator: &CanonicalAddr,
+        salt: &[u8],
+    ) -> Result<Addr> {
+        // string representation of the salt
+        let salt_to_string = std::str::from_utf8(salt)?;
+        // Remove the square brackets and split the string by commas
+        let parts: Vec<&str> = salt_to_string
+            .trim_matches(|c| c == '[' || c == ']')
+            .split(',')
+            .collect();
+        // Convert each part to a u8 and collect them into a Vec<u8>
+        let salt: Vec<u8> = parts
+            .iter()
+            .map(|part| part.trim().parse().unwrap())
+            .collect();
+        let canonical_addr = instantiate2_address(checksum, creator, &salt)?;
```

Comment:
the magic is here

Reply:
Correct, this is executed when `WasmMsg::Instantiate2` is called.